File: exam.cpp
#include <iostream>
#include <chrono>

int main()
{
    int NUM_ITERATIONS = 1000000000;
    int counter = 0;

    auto start = std::chrono::high_resolution_clock::now();

    for (int i = 0; i < NUM_ITERATIONS; ++i)
    {
        counter++;
    }
    // int i = 0;
    // while (i < NUM_ITERATIONS)
    // {
    //     counter++;
    //     i++;
    // }

    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> elapsed = end - start;

    std::cout << "Counter: " << counter << std::endl;
    std::cout << "Time taken: " << elapsed.count() << " seconds" << std::endl;

    return 0;
}

File: exam.js
const NUM_ITERATIONS = 1000000000;
let counter = 0;

const start = process.hrtime.bigint();

for (let i = 0; i < NUM_ITERATIONS; i++) {
    counter++;
}

const end = process.hrtime.bigint();
const elapsedNs = end - start;
const elapsedSeconds = Number(elapsedNs) / 1e9;

console.log(`Counter: ${counter}`);
console.log(`Time taken: ${elapsedSeconds} seconds`);
